src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -241,11 +241,12 @@ SHAMapStoreImp::makeDatabase (std::string const& name,
 }
 
 void
-SHAMapStoreImp::onLedgerClosed (Ledger::pointer validatedLedger)
+SHAMapStoreImp::onLedgerClosed(
+    std::shared_ptr<Ledger const> const& ledger)
 {
     {
         std::lock_guard <std::mutex> lock (mutex_);
-        newLedger_ = validatedLedger;
+        newLedger_ = ledger;
```

Comment:
Nit: we'd be better off taking `ledger` by value and moving. Especially given how this is called in LedgetMaster.cpp:238 (but it's hard to argue it's worthwhile given we take a `mutex_`)